compiler/modules/comm.diagnostic/diagnostic.bal

```diff
@@ -112,7 +113,43 @@ public function messageFormat(Template t) returns string {
 
 public function format(Diagnostic d) returns string[] {
     Location loc = d.location;
+    string[] lines = [];
+    string line = loc.file.filename();
     LineColumn lc = locationLineColumn(loc);
     // lineColumn returns 0-based lines currently
-    return [loc.file.filename() + ":" + lc[0].toString() + ":" + (lc[1] + 1).toString() + ": error: " + d.message];
+    line += ":" + lc[0].toString() + ":" + (lc[1] + 1).toString();
+    line += ": error: " + d.message;
+    lines.push(line);
+    Range|Position range = loc.range;
+    lines.push(...fileLines(loc.file, range));
+    return lines;
+}
+
+function fileLines(File file, Range|Position range) returns [string, string] {
+    string line = range is Position ? file.lineContent(range) : file.lineContent(range.startPos);
```

Comment:
Call lineContext in `format` and have this function return just a `string`: call it caretLine.

Reply:
fixed